packages/foam-core/test/markdown-provider.test.ts

```diff
@@ -196,6 +196,54 @@ describe('wikilinks definitions', () => {
   });
 });
 
+describe('tags plugin', () => {
+  it('can find tags in the text of the note', () => {
+    const noteA = createNoteFromMarkdown(
+      '/dir1/page-a.md',
+      `
+# this is a heading
+this is some #text that includes #tags we #care-about.
+    `,
+      '\n'
+    );
+    expect(noteA.tags).toEqual(new Set(['text', 'tags', 'care-about']));
+  });
+
+  it('can find tags as text in yaml', () => {
+    const noteA = createNoteFromMarkdown(
+      '/dir1/page-a.md',
+      `
+---
+tags: hello, world  this_is_good
```

Comment:
Does the double space here result in "world" and "this_is_good" separate tags? This is just curiousity on my part! Seems weird to me

Reply:
I actually did it on purpose :) 
I am mixing various separators in this test and wanted to validate they can be mixed (basically I don't want to be strict about the syntax of the `tags` property, people might have different conventions)